### PR TITLE
Use snapshot builder (ephemeral.go) for validation.

### PR DESCRIPTION
**Pull request by guptasu**
* **Delete existing runtime/validator code and use ephemeral to validate the config**. 
Validator is used during config ingestion via the webhook and ephemeral code is also used to build config snapshot by the runtime.
After this PR, both the codes are same. Earlier in ephemeral.go we were ignoring the errors, now we just return them back to the caller so validation can use it too.

* Also made the **validation more thorough**. During config ingestion, we should invoke adapter's SetType, SetConfig and then Validate. We were not doing this properly before inside the validator.

* Had to **refactor the factory.go from runtime/handler to runtime/config** because config needs most of it's functionality and was causing a cyclic dependency. Plus, the factory.go's 90% of the code get's run during config time.

* Moved the runtime/validator pkg to runtime/config/validator.

* **Removed the cachedAttributes from ephemeral.go.** Which mean everything there is a config change, the full snapshot including the attributes config will be processed fresh. Since this is during config back, I **hope the tiny perf impact is not important**. _Reason I had to remove the_ cache is because now the ephemeral's state (entries map) will be updated asynchronously via the webhook and via the store's callback itself. When the store updates the entries, we do not compute the attributes and as a result the cached attributes and the entires will go out of sync. After this PR, we only need to protect the entries map under the mutex, which is simple.

### CODE REVIEW

File: mixer/testdata/config/cloudwatch.yaml [hunk 1]
```diff
@@ -25,7 +25,6 @@ spec:
  metricInfo:
    requestcount.metric.istio-system:
      unit: Count
-     storageResolution: 1 #high resolution metric
```
Comment (mandarjog) [inline mixer/testdata/config/cloudwatch.yaml]: did you mean to remove this ?

File: galley/cmd/gals/cmd/validator.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func createMixerValidator(kubeconfig string) (store.BackendValidator, error) {
 	if err != nil {
 		return nil, err
 	}
-	rv, err := validator.New(checker.NewTypeChecker(), "", s, adapters, templates)
+	rv, err := validator.NewValidator(checker.NewTypeChecker(), "", s, adapters, templates)
```
Comment (mandarjog) [inline galley/cmd/gals/cmd/validator.go]: stutter? why not `New` like it was before?

File: mixer/testdata/config/cloudwatch.yaml (hunk 1 above)
Comment (guptasu) [reply mixer/testdata/config/cloudwatch.yaml]: Yes, this field does not exist

File: galley/cmd/gals/cmd/validator.go (hunk 2 above)
Comment (guptasu) [reply galley/cmd/gals/cmd/validator.go]: Will fix

**Comment by guptasu** [comment]
oops.. @mandarjog, @geeknoid and @ozevren this PR went in by the bot. @ozevren, the code that is committed has fixes for your comments, but since you were reviewing, do you prefer a revert or continue comments on this PR. I will create a separate PR to address your second round of comments . WDYT ?

**Comment by guptasu** [comment]
@mandarjog with your comment quoted below, which part were you referring to ?
_"I am not fully onboard with making functional changes to improve code re-use.
However this does already represent a good step towards eliminating overly duplicative code.
At this stage we can go ahead with this PR, and fix those issues in subsequent PRs."_

**Comment by mandarjog** [comment]
I was referring to “Removed the cachedAttributes from ephemeral.go.”

**Pull request merged**